### PR TITLE
Deploy plugin with ssl enabled

### PR DESCRIPTION
**Pull request by Msarawan**
Description:

Fixed deploying plugin with ssl enable. now the plugin is loading in the openshift dev-cluster.

Screenshot from 2024-01-25 18-13-48

Screenshot from 2024-01-25 14-53-55

### CODE REVIEW

File: webpack.config.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import { Configuration as WebpackConfiguration } from 'webpack';
 import { Configuration as WebpackDevServerConfiguration } from 'webpack-dev-server';
 import * as path from 'path';
+//import fs from 'fs';
```
Comment (RoddieKieley) [inline webpack.config.ts:6]: Was it intentional to leave the comment in for future reference?

File: http-server.sh [hunk 2]
```diff
@@ -6,4 +6,5 @@ PUBLIC_PATH="$1"
 shift
 SERVER_OPTS="$@"
 
-./node_modules/.bin/http-server $PUBLIC_PATH -p 8001 -c-1 --cors $SERVER_OPTS
\ No newline at end of file
+echo "starting server with options: $SERVER_OPTS"
+./node_modules/.bin/http-server $PUBLIC_PATH -c-1 $SERVER_OPTS
```
Comment (RoddieKieley) [inline http-server.sh:10]: Was the removal of --cors intentional here? It's not added back in the ENTRYPOINT above.

File: manifest.yaml [hunk 3]
```diff
@@ -66,35 +58,10 @@ spec:
...
-        root                /usr/share/nginx/html;
-      }
-    }
----
-apiVersion: v1
 kind: Service
 metadata:
   annotations:
-    service.alpha.openshift.io/serving-cert-secret-name: plugin-serving-cert
+    service.beta.openshift.io/serving-cert-secret-name: plugin-serving-cert
```
Comment (RoddieKieley) [inline manifest.yaml:64]: Are we eliminating compatibility with any older versions by moving from alpha to beta here? If so it's a good item to to point out via the README.md and other docs.

File: webpack.config.ts [hunk 4]
```diff
@@ -58,7 +59,12 @@ const config: Configuration = {
   },
   devServer: {
     static: './dist',
-    port: 9001,
+    port: 9443,
```
Comment (RoddieKieley) [inline webpack.config.ts:62]: If we were to change the port passed in the SERVER_OPTS would we also need to update here?

File: webpack.config.ts [hunk 5]
```diff
@@ -58,7 +59,12 @@ const config: Configuration = {
...
+    https: true,
+    // {
+    //  key: fs.readFileSync('/var/serving-cert/plugin-serving-cert/key.pem'),
+    //  cert: fs.readFileSync('/var/serving-cert/plugin-serving-cert/cert.pem'),
+    //},
```
Comment (RoddieKieley) [inline webpack.config.ts:67]: Should these be removed?

**Review by RoddieKieley — COMMENTED** [review]
A few questions and items to consider before merge

File: webpack.config.ts (hunk 1 above)
Comment (Msarawan) [reply webpack.config.ts:6]: Nope...i will update the PR by removing the comment lines.

File: webpack.config.ts (hunk 4 above)
Comment (Msarawan) [reply webpack.config.ts:62]: yes!

File: webpack.config.ts (hunk 5 above)
Comment (Msarawan) [reply webpack.config.ts:67]: yes! i will update the PR by removing the comment lines.

File: http-server.sh (hunk 2 above)
Comment (Msarawan) [reply http-server.sh:10]: nope! i will update it back.

**Review by ajaypratap003 — APPROVED** [review]

**Pull request merged**